src/profile-logic/profile-data.js

```diff
@@ -2609,15 +2609,23 @@ export function extractProfileFilterPageData(
     return null;
   }
 
-  // Getting the first page's innerWindowID and then getting its url.
-  const innerWindowID = [...relevantPages][0];
-  const filteredPages = pages.filter(
-    page => page.innerWindowID === innerWindowID
+  // Getting the pages that are relevant and a topmost frame.
+  let filteredPages = pages.filter(
+    page =>
+      relevantPages.has(page.innerWindowID) && page.embedderInnerWindowID === 0
```

Comment:
can you please explain a bit more what `embedderInnerWindowID` means? (could be either here or in `types/profile`)
In the type file, it's written `// 0 means no embedder`, I believe this isn't clear enough. Maybe replace that term "no embedder" with the concept of a "main page" or "topmost page"? Or explain that this defines a tree of pages, where `embedderInnerWindowID` points to the frame that contains this page?

Reply:
Added comments to the type we have in the `types/profile`.